File: leetcode/2756-buy-two-chocolates/buy-two-chocolates.py
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        # [1,2,2] choc prices
        # 3 = money
        # need two chocalates without having debt

        # return left over moeny, after buying 2 chocs

        prices.sort()
        new_money = money
        for i in range(2):
          new_money -= prices[i]
        
        if new_money >= 0:
          return new_money

        return money

File: leetcode/2756-buy-two-chocolates/buy-two-chocolates.cpp
class Solution {
public:
    int buyChoco(vector<int>& prices, int money) {
        auto n = prices.size();
        auto other = money;
        nth_element(prices.begin(), prices.begin(), prices.end());
        money -= prices[0], money -= prices[1];
        return money < 0 ? other : money;
    }
};